packages/rulesets/src/spectral/test/delete-response-codes.test.ts

```diff
@@ -0,0 +1,588 @@
+              description: "No content",
+              schema: {
+                $ref: "#/definitions/FooResource",
+              },
+            },
+          },
+          "x-ms-long-running-operation": true,
+        },
+      },
+    },
+    definitions: {
+      FooRequestParams: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooResource: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooResourceUpdate: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooProps: {
+        properties: {
+          prop0: {
+            type: "string",
+            default: "my def val",
+          },
+        },
+      },
+    },
+  }
+  return linter.run(myOpenApiDocument).then((results) => {
+    expect(results.length).toBe(1)
+    expect(results[0].path.join(".")).toBe("paths./foo.delete")
+    expect(results[0].message).toContain(LR_ERROR)
+  })
+})
+
+test("DeleteResponseCodes should find errors for lro delete without default response", () => {
+  const myOpenApiDocument = {
+    swagger: "2.0",
+    paths: {
+      "/foo": {
+        delete: {
+          tags: ["SampleTag"],
+          operationId: "Foo_Update",
+          description: "Test Description",
+          parameters: [
+            {
+              name: "foo_delete",
+              in: "body",
+              schema: {
+                $ref: "#/definitions/FooRequestParams",
+              },
+            },
+          ],
+          responses: {
+            "202": {
+              description: "accepted",
+              schema: {
+                $ref: "#/definitions/FooResource",
+              },
+            },
+            "204": {
+              description: "No content",
+              schema: {
+                $ref: "#/definitions/FooResource",
+              },
+            },
+          },
+          "x-ms-long-running-operation": true,
+        },
+      },
+    },
+    definitions: {
+      FooRequestParams: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooResource: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooResourceUpdate: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooProps: {
+        properties: {
+          prop0: {
+            type: "string",
+            default: "my def val",
+          },
+        },
+      },
+    },
+  }
+  return linter.run(myOpenApiDocument).then((results) => {
+    expect(results.length).toBe(1)
+    expect(results[0].path.join(".")).toBe("paths./foo.delete")
+    expect(results[0].message).toContain(LR_ERROR)
+  })
+})
+
+test("DeleteResponseCodes should find no errors for lro delete", () => {
```

Comment:
```suggestion
test("DeleteResponseCodes should find no errors for lro delete when all required codes are provided", () => {
```

I suggest similar fix to other relevant tests.